tracer/src/Datadog.Trace/Sampling/SpanSamplingRule.cs

```diff
@@ -0,0 +1,157 @@
+// <copyright file="SpanSamplingRule.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.RegularExpressions;
+using Datadog.Trace.Logging;
+using Datadog.Trace.Util;
+using Datadog.Trace.Vendors.Newtonsoft.Json;
+
+namespace Datadog.Trace.Sampling
+{
+    /// <summary>
+    /// Represents a sampling rules for single span ingestion.
+    /// </summary>
+    internal class SpanSamplingRule : ISpanSamplingRule
+    {
+        private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor<SpanSamplingRule>();
+        private static readonly TimeSpan RegexTimeout = TimeSpan.FromSeconds(1);
+
+        // TODO consider moving toward this https://github.com/dotnet/runtime/blob/main/src/libraries/Common/src/System/Text/SimpleRegex.cs
+        private readonly Regex _serviceNameRegex;
+        private readonly Regex _operationNameRegex;
+
+        private readonly IRateLimiter _limiter;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="SpanSamplingRule"/> class with default values.
+        /// </summary>
+        public SpanSamplingRule()
+        {
+        }
```

Comment:
It doesn't seem right to have this rule, which leaves the regexes etc `null`, but then throw in the other constructor. Do you _need_ this constructor? Can we remove it?